### PR TITLE
Try to improve builds

### PR DESCRIPTION
**Pull request by ryanthemanuel**
### Summary
Try and improve the builds with caching.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,46 +5,66 @@ services:
...
+      script:
+        - 'cp ./config/docker/ci-docker-compose.yml ./docker-compose.yml'
+        - travis_wait docker-compose build --build-arg TERRA_NODE_TAG=latest test-chrome
+        - travis_wait docker-compose --verbose pull standalone-chrome
+        - >
+          mkdir -p $HOME/docker && docker images -a --filter='dangling=false' --format '{{.Repository}}:{{.Tag}} {{.ID}}'
+          | xargs -n 2 -t sh -c 'test -e $HOME/docker/$1.tar.gz || docker save $0 | gzip -2 > $HOME/docker/$1.tar.gz'
 stages:
+  - name: release terra-node
+    if: type = pull_request
```
Comment (mjhenkes) [inline .travis.yml]: does this mean we only build and test for pull-requests? and not when we build the master branch?

Comment (mjhenkes) [reply .travis.yml]: actually this should probably only be in terra-toolkit since thats where terra-node lives

File: .travis.yml [hunk 2]
```diff
@@ -46,5 +79,9 @@ jobs:
           branch: master
 stages:
   - build
+  - lint
+  - jest
   - test
-  - deploy
+  - danger
```
Comment (mjhenkes) [inline .travis.yml]: Could we move danger above test? Its another fail fast step.

**Comment by ryanthemanuel** [comment]
Yes to Brett’s question

**Pull request merged**